gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoRestSerializer.java

```diff
@@ -118,6 +148,18 @@ public void putPathParam(Map<String, String> fields, String fieldName, Object fi
     fields.put(fieldName, String.valueOf(fieldValue));
   }
 
+  private void putDecomposedMessageQueryParam(
+      Map<String, List<String>> fields, String fieldName, Object fieldValue) {
+    for (Map.Entry<FieldDescriptor, Object> fieldEntry :
+        ((GeneratedMessageV3) fieldValue).getAllFields().entrySet()) {
```

Comment:
This cast is only safe due to an instanceof check that occurs elsewhere in the logic. Will the next maintainer know they must call `isNonSerializableMessageValue()` as a check before this method?

I would highly suggest either performing the check explicitly in this method, or changing the parameter from `Object` to `GeneratedMessageV3`, thus forcing the caller of this method to perform the cast.

In general, the type-check (`instanceof`) and type-assertion (`cast`) should always be colocated - as two actions are tightly coupled and will need to change together if one is modified.